File: AbstractFactoryDesignPattern/GUIComponentCreator_Example/Factory/WindowsFactory.java
package AbstractFactoryDesignPattern.GUIComponentCreator_Example.Factory;

import AbstractFactoryDesignPattern.GUIComponentCreator_Example.Product.Button;
import AbstractFactoryDesignPattern.GUIComponentCreator_Example.Product.TextBox;
import AbstractFactoryDesignPattern.GUIComponentCreator_Example.Product.WindowsButton;
import AbstractFactoryDesignPattern.GUIComponentCreator_Example.Product.WindowsTextBox;

public class WindowsFactory implements Factory {
	@Override
	public Button createButton() {
		return new WindowsButton();
	}

	@Override
	public TextBox createTextBox() {
		return new WindowsTextBox();
	}
}

File: AbstractFactoryDesignPattern/GUIComponentCreator_Example_Python/Factory/WindowsFactory.py
from AbstractFactoryDesignPattern.GUIComponentCreator_Example_Python.Factory.FactoryInterface import FactoryInterface
from AbstractFactoryDesignPattern.GUIComponentCreator_Example_Python.Product.Button import Button
from AbstractFactoryDesignPattern.GUIComponentCreator_Example_Python.Product.TextBox import TextBox
from AbstractFactoryDesignPattern.GUIComponentCreator_Example_Python.Product.WindowsButton import WindowsButton
from AbstractFactoryDesignPattern.GUIComponentCreator_Example_Python.Product.WindowsTextBox import WindowsTextBox


class WindowsFactory(FactoryInterface):

	def createButton(self) -> Button:
		return WindowsButton()

	def createTextBox(self) -> TextBox:
		return WindowsTextBox()
